addons/loyalty/tests/test_loyalty.py

```diff
@@ -125,3 +126,23 @@ def test_archiving_unarchiving(self):
         self.program.toggle_active()
         after_archived_reward_ids = self.program.reward_ids
         self.assertEqual(before_archived_reward_ids, after_archived_reward_ids)
+
+    def test_prevent_archive_loyalty_product_linked_active_loyalty_reward(self):
```

Comment:
Suggestion here, I'm pretty bad at naming my test methods too '^^
```suggestion
    def test_prevent_archiving_product_linked_to_active_loyalty_reward(self):
```